### PR TITLE
Useful error messages

### PR DESCRIPTION
**Pull request by gabriel090**
Useful error messages instead of silent errors e.g. when a user doesn't have permissions to execute some function it should display useful errors

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Screenshots

Images before error
order_before

images after error was rectified
order_after

### CODE REVIEW

**Comment by ojwanganto** [comment]
@gabriel090 - you may need to rebase the codebase and update the PR. Kindly change the title to be clearer and link this to a ticket in OpenMRS jira. Thanks

File: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
Comment (denniskigen) [inline .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs:1]: I wouldn't expect to see changes to this file. Did it get deleted by mistake?

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-nodeLinker: node-modules
+compressionLevel: mixed
+
+enableGlobalCache: false
 
-plugins:
-  - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
-    spec: "@yarnpkg/plugin-interactive-tools"
+nodeLinker: node-modules
```
Comment (denniskigen) [inline .yarnrc.yml:6]: Or here.

File: packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.workspace.tsx [hunk 2]
```diff
@@ -54,9 +58,24 @@ export default function AddDrugOrderWorkspace({ order: initialOrder, closeWorksp
...
+    [orders, setOrders, closeWorkspace, session?.currentProvider?.uuid],
   );
 
+  if (session.currentProvider == null) {
+    return (
+      <div className={styles.tileContainer}>
+        <Tile className={styles.tile}>
+          <div className={styles.tileContent}>
+            <p className={styles.content}>
+              {t('providerNotAllowed', 'User provider is not allowed to perform this action')}
```
Comment (denniskigen) [inline packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.workspace.tsx:70]: @ibacher how would you word this better?

Also, @gabriel090, the established convention here is to render one of the [notification types](https://zeroheight.com/23a080e38/p/683580-notifications/t/60ea14) described in the notifications guide.

File: yarn.lock
Comment (denniskigen) [inline yarn.lock:1]: I wouldn't expect to see changes to the lockfile without a corresponding change to the `package.json` files of either the labs or medications apps.

**Review by denniskigen — CHANGES_REQUESTED** [review]

File: packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.workspace.tsx (hunk 2 above)
Comment (ibacher) [reply packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.workspace.tsx:70]: "The current user is not setup as a provider, and so cannot place drug orders. If you feel that this is in error, please contact your system administrator for assistance."

Comment (ibacher) [reply packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.workspace.tsx:70]: Also, I would use something like the same text for lab orders. This _should_ be a toast notification.

**Review by ibacher — CHANGES_REQUESTED** [review]
As @denniskigen's comments indicate, please revert the Yarn packaging changes and move any updates to the Yarn version to  a separate PR.

File: .yarnrc.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
-nodeLinker: node-modules
+compressionLevel: mixed
+
```
Comment (suubi-joshua) [inline .yarnrc.yml:2]: @gabriel090 are these spaces required?

**Comment by denniskigen** [comment]
Closing as stale.

**Pull request closed without merging**